test/car_req_test.exs

```diff
@@ -756,6 +756,53 @@ defmodule CarReqTest do
                       %{datadog_service_name: :guu_car_caz}}
     end
 
+    test "allows client to set a resource_name_override rule" do
+      defmodule ResourceNameOverrideClient do
+        use CarReq,
+          resource_name_override: &ResourceNameOverrideClient.resource_name_override/1
+
+        def resource_name_ovveride(resource) do
+          String.replace(resource, ~r|\d+|, "{guid}")
+        end
+      end
+
+      ResourceNameOverrideClient.request(method: :get, url: "/200", adapter: &Adapter.success/1)
+
+      assert_receive {:event, [:http_car_req, :request, :start], _,
+                      %{resource_name_override: override}}
+
+      assert is_function(override, 1)
+    end
+
+    test "allows resource name override option to be set per-request on the client" do
+      defmodule ResourceNameOverrideRuntimeClient do
+        use CarReq,
+          resource_name_override: &ResourceNameOverrideRuntimeClient.global_override/1
+
+        def request_override(resource_name) do
+          String.replace(resource_name, ~r|\d+|, "{guid}")
+        end
+
+        def global_override(resource_name) do
+          String.replace(resource_name, "/", "<slash>")
+        end
+      end
+
+      url = "/employees/1234/details"
+
+      ResourceNameOverrideRuntimeClient.request(
+        method: :get,
+        url: url,
+        adapter: &Adapter.success/1,
+        resource_name_override: &ResourceNameOverrideRuntimeClient.request_override/1
+      )
```

Comment:
at the point of the request, does this need to be a function or could this also accept a string?

Reply:
I did not even consider that, but yes it could be and that probably makes sense in a lot of cases (i.e., we want the resource name to be static rather than dynamic). I'll make a change to slip this in too.

Reply:
Nice, I think a test showing both the fn call and a static string would make sense.

Reply:
Just added a test case and expanded the docs example to show the static string usage.